Bottom: e383550b1251e229ced47388029e4b7ea760f5ee
Top:    94b8d375e1020f4429add2976c1537a894854f30
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-18 21:54:34 +0100

Refresh of initialise-the-hid-vtable-by-f

---

diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index b2338cb..b11fb8b 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -347,63 +347,63 @@ batch_show_item (void *item)
 
 #include "dolists.h"
 
-HID batch_gui;
+static HID batch_hid;
 
 void
 hid_batch_init ()
 {
-  memset (&batch_gui, 0, sizeof (HID));
-
-  batch_gui.struct_size           = sizeof (HID);
-  batch_gui.name                  = "batch";
-  batch_gui.description           = "Batch-mode GUI for non-interactive use.";
-
-  batch_gui.gui                   = 1;
-  batch_gui.printer               = 0;
-  batch_gui.exporter              = 0;
-  batch_gui.poly_before           = 0;
-  batch_gui.poly_after            = 0;
-  batch_gui.poly_dicer            = 0;
-
-  batch_gui.get_export_options    = batch_get_export_options;
-  batch_gui.do_export             = batch_do_export;
-  batch_gui.parse_arguments       = batch_parse_arguments;
-  batch_gui.invalidate_lr         = batch_invalidate_lr;
-  batch_gui.invalidate_all        = batch_invalidate_all;
-  batch_gui.set_layer             = batch_set_layer;
-  batch_gui.make_gc               = batch_make_gc;
-  batch_gui.destroy_gc            = batch_destroy_gc;
-  batch_gui.use_mask              = batch_use_mask;
-  batch_gui.set_color             = batch_set_color;
-  batch_gui.set_line_cap          = batch_set_line_cap;
-  batch_gui.set_line_width        = batch_set_line_width;
-  batch_gui.set_draw_xor          = batch_set_draw_xor;
-  batch_gui.set_draw_faded        = batch_set_draw_faded;
-  batch_gui.set_line_cap_angle    = batch_set_line_cap_angle;
-  batch_gui.draw_line             = batch_draw_line;
-  batch_gui.draw_arc              = batch_draw_arc;
-  batch_gui.draw_rect             = batch_draw_rect;
-  batch_gui.fill_circle           = batch_fill_circle;
-  batch_gui.fill_polygon          = batch_fill_polygon;
-  batch_gui.fill_pcb_polygon      = batch_fill_pcb_polygon;
-  batch_gui.thindraw_pcb_polygon  = batch_thindraw_pcb_polygon;
-  batch_gui.fill_rect             = batch_fill_rect;
-  batch_gui.calibrate             = batch_calibrate;
-  batch_gui.shift_is_pressed      = batch_shift_is_pressed;
-  batch_gui.control_is_pressed    = batch_control_is_pressed;
-  batch_gui.mod1_is_pressed       = batch_mod1_is_pressed;
-  batch_gui.get_coords            = batch_get_coords;
-  batch_gui.set_crosshair         = batch_set_crosshair;
-  batch_gui.add_timer             = batch_add_timer;
-  batch_gui.stop_timer            = batch_stop_timer;
-  batch_gui.watch_file            = batch_watch_file;
-  batch_gui.unwatch_file          = batch_unwatch_file;
-  batch_gui.add_block_hook        = batch_add_block_hook;
-  batch_gui.stop_block_hook       = batch_stop_block_hook;
-  batch_gui.attribute_dialog      = batch_attribute_dialog;
-  batch_gui.show_item             = batch_show_item;
-
-  apply_default_hid (&batch_gui, 0);
-  hid_register_hid (&batch_gui);
+  memset (&batch_hid, 0, sizeof (HID));
+
+  batch_hid.struct_size           = sizeof (HID);
+  batch_hid.name                  = "batch";
+  batch_hid.description           = "Batch-mode GUI for non-interactive use.";
+
+  batch_hid.gui                   = 1;
+  batch_hid.printer               = 0;
+  batch_hid.exporter              = 0;
+  batch_hid.poly_before           = 0;
+  batch_hid.poly_after            = 0;
+  batch_hid.poly_dicer            = 0;
+
+  batch_hid.get_export_options    = batch_get_export_options;
+  batch_hid.do_export             = batch_do_export;
+  batch_hid.parse_arguments       = batch_parse_arguments;
+  batch_hid.invalidate_lr         = batch_invalidate_lr;
+  batch_hid.invalidate_all        = batch_invalidate_all;
+  batch_hid.set_layer             = batch_set_layer;
+  batch_hid.make_gc               = batch_make_gc;
+  batch_hid.destroy_gc            = batch_destroy_gc;
+  batch_hid.use_mask              = batch_use_mask;
+  batch_hid.set_color             = batch_set_color;
+  batch_hid.set_line_cap          = batch_set_line_cap;
+  batch_hid.set_line_width        = batch_set_line_width;
+  batch_hid.set_draw_xor          = batch_set_draw_xor;
+  batch_hid.set_draw_faded        = batch_set_draw_faded;
+  batch_hid.set_line_cap_angle    = batch_set_line_cap_angle;
+  batch_hid.draw_line             = batch_draw_line;
+  batch_hid.draw_arc              = batch_draw_arc;
+  batch_hid.draw_rect             = batch_draw_rect;
+  batch_hid.fill_circle           = batch_fill_circle;
+  batch_hid.fill_polygon          = batch_fill_polygon;
+  batch_hid.fill_pcb_polygon      = batch_fill_pcb_polygon;
+  batch_hid.thindraw_pcb_polygon  = batch_thindraw_pcb_polygon;
+  batch_hid.fill_rect             = batch_fill_rect;
+  batch_hid.calibrate             = batch_calibrate;
+  batch_hid.shift_is_pressed      = batch_shift_is_pressed;
+  batch_hid.control_is_pressed    = batch_control_is_pressed;
+  batch_hid.mod1_is_pressed       = batch_mod1_is_pressed;
+  batch_hid.get_coords            = batch_get_coords;
+  batch_hid.set_crosshair         = batch_set_crosshair;
+  batch_hid.add_timer             = batch_add_timer;
+  batch_hid.stop_timer            = batch_stop_timer;
+  batch_hid.watch_file            = batch_watch_file;
+  batch_hid.unwatch_file          = batch_unwatch_file;
+  batch_hid.add_block_hook        = batch_add_block_hook;
+  batch_hid.stop_block_hook       = batch_stop_block_hook;
+  batch_hid.attribute_dialog      = batch_attribute_dialog;
+  batch_hid.show_item             = batch_show_item;
+
+  apply_default_hid (&batch_hid, 0);
+  hid_register_hid (&batch_hid);
 #include "batch_lists.h"
 }
